### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by tommylettieri**
Added nodemon for quicker auto restarts and quicker deployment
Added response formatting to limiter
Added failure case to account verification
Added linter
Fixed all linting issues
Fixed register call to not return sensitive cases
Added awaits for startup to avoid errors
Added permission model
Added permission logic on register

### CODE REVIEW

File: src/database/index.ts [hunk 1]
```diff
@@ -4,19 +4,22 @@ import appSequelize from './app-sequelize';
...
 import session from './models/session';
+import permission from './models/permission';
 
-(async () => {
+export const sync = async (): Promise<void> => {
     try {
-        appSequelize.authenticate();
-        appSequelize.sync()    
+        await appSequelize.authenticate();
+        await appSequelize.sync()
```
Comment (gbhojraj) [inline src/database/index.ts:12]:
```suggestion
        await appSequelize.sync();
```

File: src/features/users/user-controller.ts [hunk 2]
```diff
@@ -115,24 +116,33 @@ class UserController {
         if (universities.length > 1) {
             logger.error(`Multiple universities found ${universities.length}`);
         }
-        university = universities[0];
+        const university = universities[0];
 
 
         if (university.student_email_domain === emailDomain) {
-            logger.info('User is student');
+            // Database field
```
Comment (gbhojraj) [inline src/features/users/user-controller.ts:123]: Logger statements could still be useful, no?

File: src/features/users/user-route.ts [hunk 3]
```diff
@@ -15,8 +15,10 @@ import AlreadyExistsError from "../../exceptions/already-exists-error";
 router.post('/login',
     validate(loginValidation),
     passport.authenticate('local'),
-    asyncHandler(async (req: any, res: Response, next: NextFunction) => {
-        const newSession = req.session.passport.user;
+    asyncHandler(async (req: Request, res: Response, next: NextFunction) => {
+        // TODO fix Request object for session
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (gbhojraj) [inline src/features/users/user-route.ts:20]: Do you still need this disable-line?

File: src/middleware/joi-validator.ts [hunk 4]
```diff
@@ -1,26 +1,33 @@
...
 
     // Using Request from express I get typescript errors
-    return function validateRequest(req: any, res: Response, next: any) {
-        let toValidate: any = {};
+    return function validateRequest(req: Request, res: Response, next: NextFunction): void {
+        // This is a generic callback which can be used for any object, thus this can be any as well
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const toValidate: any = {};
         if (!schema) {
             next();
```
Comment (gbhojraj) [inline src/middleware/joi-validator.ts:15]: Should this be `return`ed?

File: src/features/users/user-controller.ts (hunk 2 above)
Comment (tommylettieri) [reply src/features/users/user-controller.ts:123]: Seems like bloat when it is in the response

File: src/middleware/joi-validator.ts (hunk 4 above)
Comment (tommylettieri) [reply src/middleware/joi-validator.ts:15]: Good call, doesn't break but it probably can cause an issue

File: src/features/users/user-route.ts (hunk 3 above)
Comment (tommylettieri) [reply src/features/users/user-route.ts:21]: Looks like it from here, but not from the source, maybe this is stale

**Pull request merged**